skippy-core/src/main/java/io/skippy/core/SkippyTestApi.java

```diff
@@ -48,6 +50,36 @@ public final class SkippyTestApi {
     private final SkippyConfiguration skippyConfiguration;
     private final Map<String, Prediction> predictions = new ConcurrentHashMap<>();
 
+    /**
+     * Stack that keeps track of the execution data across nested test classes.
+     *
+     * Example:
+     * <pre>
+     * {@literal @}PredictWithSkippy
+     *  public class Level1 {
+     *     {@literal @}Nested
+     *      class Level2 {
+     *         {@literal @}Nested
+     *          class Level3 {
+     *             {@literal @}Test
+     *              void testSomething() {
+     *              }
+     *         }
+     *     }
+     * }
+     * </pre>
+     *
+     *  By the time <code>testSomething</code> is executed, the stack would look as follows:
+     *  <pre>
+     *  { execution data for Level1$Level2$Level3.class }
+     *  { execution data for Level1$Level2.class }
+     *  { execution data for Level1.class }
+     *  </pre>
+     *
+     *  Nested test classes contribute their execution data "back" to their parents using this stack.
+     */
+    private final Stack<List<byte[]>> executionDataStack = new Stack<>();
```

Comment:
Very elegant and simple solution indeed.

Reply:
I will incorporate your feedback and add more tests (including functional tests in skippy-functional-tests). 

ETA for this fixed to be released and available in main: Some time weekend.

Reply:
Lovely, thanks!